0-bootstrap/modules/jenkins/README.md

```diff
@@ -0,0 +1,134 @@
+## Overview
+
+The objective of this module is to deploy a Google Cloud Platform project `prj-cicd` to host a Jenkins Agent that can be used to deploy your infrastructure changes. This module is a replica of the [cloudbuild module](https://github.com/terraform-google-modules/terraform-google-bootstrap/tree/master/modules/cloudbuild), but re-purposed to use jenkins instead of Cloud Build. This module creates:
+- The `prj-cicd` project to hold the Jenkins Agent
+- GCE Instance for the Jenkins Agent, assigning SSH public keys in the metadata to allow connectivity from the Jenkins Master.
+- FW rules to allow communication over port 22
+  - TODO: use a fixed IP or no public IP at all
+-  Custom Service account to run the Jenkins Agent's GCE Instance
+
+## Usage
+
+1. Add the SSH public keys of your Jenkins Agent in the file `./jenkins-agent-ssh-pub-keys/metadata-ssh-pub-keys`
+
+1. Run `$ gcloud auth application-default login` before running `$ terraform plan` to avoid the errors below:
```

Comment:
I assume in a real deployment we will not rely on user credentials for terraform auth?

Reply:
No, because this module is meant to run as part of the pipeline and that will have its oun Service Account. But for the momment, I am running the module manually from my laptop, so I need to run `$ gcloud auth application-default login` fisrt. This is the same for the 0-bootstrap module by the way.

Reply:
@morgante @rjerrems what are your thoughts on this? I dont see any explicit guidance around what credentials we use for [bootstrap](https://github.com/terraform-google-modules/terraform-example-foundation/blob/develop/0-bootstrap/README.md).

Reply:
@caleonardo Sean confirmed that the seed project will need to be created manually using a pre-existing SA.

Reply:
@langleyce Why? There's no reason to use a service account for creating the bootstrap project (which is done manually). We should use application default credentials for that.

Reply:
Solving this now. Assuming we agree on using `$ gcloud auth application-default login` when running the first manual steps in 0-bootstrap.